pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go

```diff
@@ -52,10 +54,12 @@ type StackConfigPolicyList struct {
 }
 
 type StackConfigPolicySpec struct {
-	ResourceSelector metav1.LabelSelector          `json:"resourceSelector,omitempty"`
-	SecureSettings   []commonv1.SecretSource       `json:"secureSettings,omitempty"`
-	Elasticsearch    ElasticsearchConfigPolicySpec `json:"elasticsearch,omitempty"`
-	Kibana           KibanaConfigPolicySpec        `json:"kibana,omitempty"`
+	ResourceSelector metav1.LabelSelector `json:"resourceSelector,omitempty"`
+	// SecureSettings applies only to Elasticsearch
+	// TODO: do we deprecate this and remove it in the future ?
```

Comment:
FWIW it should be possible to send a warning to the user when the resource is validated, with something along the lines of:

```diff
diff --git a/pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go b/pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go
index 475ae15c8..56165c67e 100644
--- a/pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go
+++ b/pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go
@@ -56,7 +56,7 @@ type StackConfigPolicyList struct {
 type StackConfigPolicySpec struct {
        ResourceSelector metav1.LabelSelector `json:"resourceSelector,omitempty"`
        // SecureSettings applies only to Elasticsearch
-       // TODO: do we deprecate this and remove it in the future ?
+       // Deprecated: SecureSettings only applies to Elasticsearch and will be removed in a next release. It must be set per application instead.
        SecureSettings []commonv1.SecretSource       `json:"secureSettings,omitempty"`
        Elasticsearch  ElasticsearchConfigPolicySpec `json:"elasticsearch,omitempty"`
        Kibana         KibanaConfigPolicySpec        `json:"kibana,omitempty"`
diff --git a/pkg/apis/stackconfigpolicy/v1alpha1/webhook.go b/pkg/apis/stackconfigpolicy/v1alpha1/webhook.go
index 2ca16658a..f73b0f455 100644
--- a/pkg/apis/stackconfigpolicy/v1alpha1/webhook.go
+++ b/pkg/apis/stackconfigpolicy/v1alpha1/webhook.go
@@ -5,6 +5,8 @@
 package v1alpha1
 
 import (
+       "fmt"
+
        apierrors "k8s.io/apimachinery/pkg/api/errors"
        "k8s.io/apimachinery/pkg/runtime"
        "k8s.io/apimachinery/pkg/runtime/schema"
@@ -36,6 +38,20 @@ var (
 
 var _ webhook.Validator = &StackConfigPolicy{}
 
+const (
+       SpecSecureSettingsDeprecated = "spec.SecureSettings is deprecated, secure settings must be set per application"
+)
+
+func (s *StackConfigPolicy) GetWarnings() []string {
+       if s == nil {
+               return nil
+       }
+       if len(s.Spec.SecureSettings) > 0 {
+               return []string{fmt.Sprintf("%s %s/%s: %s", Kind, s.Namespace, s.Name, SpecSecureSettingsDeprecated)}
+       }
+       return nil
+}
+
 // ValidateCreate is called by the validating webhook to validate the create operation.
 // Satisfies the webhook.Validator interface.
 func (p *StackConfigPolicy) ValidateCreate() (admission.Warnings, error) {
```

Not tested, but we do something similar for Agent: https://github.com/elastic/cloud-on-k8s/blob/5937685b9085a5b185dde3d31ad50ccb5cdc034b/pkg/apis/agent/v1alpha1/webhook.go#L37-L45